src/object_store.cpp

```diff
@@ -786,7 +786,35 @@ void ObjectStore::apply_schema_changes(Group& group, uint64_t schema_version,
                                        util::Optional<std::string> sync_user_id,
                                        std::function<void()> migration_function)
 {
-    create_metadata_tables(group, sync_user_id != util::none);
+    // Temporary fix for https://github.com/realm/realm-object-store/pull/698
```

Comment:
Rather than doing all this you can just remove the call to ` _impl::initialize_schema()` in `create_metadata_tables()` and add it after applying the changes from the user-specified schema.

Reply:
Ah, much better idea than this 👍